src/plugins/colum-chooser/column-chooser.component.ts

```diff
@@ -0,0 +1,231 @@
+import {Component, Optional, OnInit, OnDestroy, Input, Output, EventEmitter, ChangeDetectorRef} from '@angular/core';
+import {Command} from 'ng2-qgrid/core/command';
+import {PipeUnit} from 'ng2-qgrid/core/pipe/pipe.unit2';
+import {GridService} from 'ng2-qgrid/main/grid';
+import * as columnService from 'ng2-qgrid/core/column/column.service';
+import {isFunction, noop} from 'ng2-qgrid/core/utility';
+import {Aggregation} from 'ng2-qgrid/core/services';
+
+import {RootService} from 'ng2-qgrid/infrastructure/component';
+import {PluginComponent} from '../plugin.component';
+import {TemplatePath} from 'ng2-qgrid/core/template';
+
+const GRID = 'qGrid';
+const COLUMN_CHOOSER_NAME = `${GRID}ColumnChooser`;
+
+@Component({
+	selector: 'q-grid-column-chooser',
+	template: `
+		<ng-container key="column-chooser.tpl" [context]="context"></ng-container>
+	`
+	// templateUrl: './column-chooser.tpl.html'
+})
+export class ColumnChooserComponent extends PluginComponent implements OnInit, OnDestroy {
+	@Input('canAggregate') columnChooserCanAggregate: boolean;
+	@Output('submit') submitEvent = new EventEmitter<any>();
+	@Output('cancel') cancelEvent = new EventEmitter<any>();
+
+	private service: any;
+	private aggregations: any;
+	private origin: any;
+	private filter: string = '';
+
+	private columns: any[];
+
+	constructor(@Optional() root: RootService, private gridService: GridService, private cdRef: ChangeDetectorRef) {
+		super(root);
+
+		this.models = ['columnChooser'];
+		this.initColumns();
+		this.defaultSelected = true;
+	}
+
+	private get columSelectors(): any[] {
+		return this.columns.filter(column => {
+			return column.title && column.key.includes(this.filter);
```

Comment:
strange code

Reply:
columnSelectors is a dynamically filtered array of the checkbox list while a user is typing into the 'Search' input. And this works just fine. What is so strange here?